packages/react-components/src/components/MessageThread.test.tsx

```diff
@@ -225,3 +227,258 @@ describe('Message should display inline image correctly', () => {
+        senderDisplayName: 'Robert Tolbert',
+        messageId: generateGUID(),
+        content: 'Absolutely! What seems to be the problem?',
+        createdOn: new Date('2019-04-13T00:00:00.000+08:11'),
+        mine: true,
+        attached: false,
+        contentType: 'html'
+      }
+    ];
+
+    const { container } = render(<MessageThread userId={user2Id} messages={messages} />);
+
+    expect(container.querySelector(`#${user2Id}`)?.nodeName.toLowerCase()).toEqual(MSFT_MENTION);
+    expect(container.querySelector(`#${user2Id}`)?.textContent).toEqual(user2Name);
+  });
+
+  test('Edited Message should include two Mentions', async () => {
+    const user1Id = 'user1';
+    const user2Id = 'user2';
+    const user3Id = 'user3';
+
+    const user2Name = 'Robert Tolbert';
+    const user3Name = 'Sam Fisher';
+
+    const messages: ChatMessage[] = [
+      {
+        messageType: 'chat',
+        senderId: user1Id,
+        senderDisplayName: 'Kat Larsson',
+        messageId: generateGUID(),
+        content: `Hey <msft-mention id="${user2Id}" displayText="${user2Name}">${user2Name}</msft-mention>, can you help me with my internet connection?`,
+        createdOn: new Date('2019-04-13T00:00:00.000+08:10'),
+        mine: false,
+        attached: false,
+        status: 'seen' as MessageStatus,
+        contentType: 'html'
+      },
+      {
+        messageType: 'chat',
+        senderId: user2Id,
+        senderDisplayName: 'Robert Tolbert',
+        messageId: generateGUID(),
+        content: 'Absolutely! What seems to be the problem?',
+        createdOn: new Date('2019-04-13T00:00:00.000+08:11'),
+        mine: true,
+        attached: false,
+        contentType: 'html'
+      }
+    ];
+
+    const { container, rerender } = render(<MessageThread userId={user2Id} messages={messages} />);
+
+    expect(container.querySelector(`#${user2Id}`)?.nodeName.toLowerCase()).toEqual(MSFT_MENTION);
+    expect(container.querySelector(`#${user2Id}`)?.textContent).toEqual(user2Name);
+
+    // edit message
+    const message1ContentAfterEdit = `Hey <msft-mention id="${user2Id}" displayText="${user2Name}">${user2Name}</msft-mention> and <msft-mention id="${user3Id}" displayText="${user3Name}">${user3Name}</msft-mention>, can you help me with my internet connection?`;
+    messages[0].content = message1ContentAfterEdit;
+    messages[0].editedOn = new Date('2019-04-13T00:01:00.000+08:10');
+    const expectedOnRenderMentionCount = 2;
+    let onRenderMentionCount = 0;
+
+    rerender(
+      <MessageThread
+        userId={user2Id}
+        messages={messages}
+        mentionOptions={{
+          displayOptions: {
+            onRenderMention: (mention, defaultOnMentionRender) => {
+              onRenderMentionCount++;
+              return <span key={generateGUID()}>{defaultOnMentionRender(mention)}</span>;
+            }
+          }
+        }}
+      />
+    );
+
+    expect(onRenderMentionCount).toEqual(expectedOnRenderMentionCount);

```

Comment:
And stronger here to compare the IDs captured to the expected values

Reply:
I think comparing the id doesn't really add much value, since the message.content will also have the id. So if we test for Id we can't tell since it's same before and after the onRenderMention render